core/src/main/scala/org/apache/spark/shuffle/ShuffleManager.scala

```diff
@@ -48,7 +48,8 @@ private[spark] trait ShuffleManager {
       handle: ShuffleHandle,
       startPartition: Int,
       endPartition: Int,
-      context: TaskContext): ShuffleReader[K, C]
+      context: TaskContext,
+      metrics: ShuffleMetricsReporter): ShuffleReader[K, C]
```

Comment:
IIUC, we should pass a read metrics reporter here, as this method is `getReader` which is called by the reducers to read shuffle files.

Reply:
It is a read metrics here actually. In the write PR this is renamed ShuffleReadMetricsReporter.